File: ccc2015s2.py
# Solution by Srihari Vishnu
#Try every jersey
j = input() #Jerseys
a = input() #Athletes

jerseys = []
requests = []
taken = [False for x in xrange(j)]
for x in range(j):
  jerseys.append(raw_input())

for y in range(a):
  requests.append(raw_input().split())
  requests[y][1] = int(requests[y][1])

count = 0
for r in requests:
  number = r[1]
  sizewanted = r[0]
  if sizewanted>=jerseys[number-1] and not taken[number-1]:
    count += 1
    taken[number-1] = True
print count

File: ccc2015s2.cpp
//Solution by Srihari Vishnu
//CCC 2015 S2 : Jerseys

#include <iostream>
#include <algorithm>
#include <string>
using namespace std;
char s;
string sizes = "SML";
int jerseys[1000001], j, a, x;
int main() {
  scanf("%d%d", &j, &a);
  for (int i = 1; i < j+1; i++) {
    scanf(" %c", &s);
    jerseys[i] = sizes.find(s);
  }
  int ans = 0;
  while (a--) {
    scanf(" %c%d", &s, &x);
    if (jerseys[x] >= int(sizes.find(s))) {
      jerseys[x] = -1;
      ans++;
    }
  }
  printf("%d", ans);

}
